opencdx-health/src/main/java/cdx/opencdx/health/model/OpenCDXAnalysisEngineModel.java

```diff
@@ -0,0 +1,145 @@
+
+    @CreatedBy
+    private OpenCDXIdentifier creator;
+
+    @LastModifiedBy
+    private OpenCDXIdentifier modifier;
+
+    /**
+     * Constructor from protobuf message AnalysisEngine
+     * @param analysisEngine Protobuf message to generate from
+     */
+    public OpenCDXAnalysisEngineModel(AnalysisEngine analysisEngine) {
+        if (analysisEngine.hasId()) {
+            this.id = new OpenCDXIdentifier(analysisEngine.getId());
+        }
+
+        this.name = analysisEngine.getName();
+        this.organizationId = analysisEngine.getOrganizationId();
+        this.workspaceId = analysisEngine.getWorkspaceId();
+
+        if (analysisEngine.hasCreated()) {
+            this.created = Instant.ofEpochSecond(
+                    analysisEngine.getCreated().getSeconds(),
+                    analysisEngine.getCreated().getNanos());
+        }
+        if (analysisEngine.hasModified()) {
+            this.modified = Instant.ofEpochSecond(
+                    analysisEngine.getModified().getSeconds(),
+                    analysisEngine.getModified().getNanos());
+        }
+        if (analysisEngine.hasCreator()) {
+            this.creator = new OpenCDXIdentifier(analysisEngine.getCreator());
+        }
+        if (analysisEngine.hasModifier()) {
+            this.modifier = new OpenCDXIdentifier(analysisEngine.getModifier());
+        }
+    }
+
+    /**
+     * Method to get the protobuf analysisEngine object
+     * @return protobuf temperature object
+     */
+    public AnalysisEngine getProtobufMessage() {
+        AnalysisEngine.Builder builder = AnalysisEngine.newBuilder();
+
+        if (this.id != null) {
+            builder.setId(this.id.toHexString());
+        }
+
+        builder.setName(this.name);
+        builder.setOrganizationId(this.organizationId);
+        builder.setWorkspaceId(this.workspaceId);
+
+        if (this.created != null) {
+            builder.setCreated(Timestamp.newBuilder()
+                    .setSeconds(this.created.getEpochSecond())
+                    .setNanos(this.created.getNano())
+                    .build());
+        }
+        if (this.modified != null) {
+            builder.setModified(Timestamp.newBuilder()
+                    .setSeconds(this.modified.getEpochSecond())
+                    .setNanos(this.modified.getNano())
+                    .build());
+        }
+        if (this.creator != null) {
+            builder.setCreator(this.creator.toHexString());
+        }
+        if (this.modifier != null) {
+            builder.setModifier(this.modifier.toHexString());
+        }
+        return builder.build();
+    }
+
+    /**
+     * Updates the OpenCDXAnalysisEngineModel with the given OpenCDXAnalysisEngine object.
+     *
+     * @param analysisEngine The analysisEngine object to update the model with.
+     * @return The updated OpenCDXTemperatureMeasurementModel instance.
+     */
+    public OpenCDXAnalysisEngineModel update(AnalysisEngine analysisEngine) {
```

Comment:
This should only update the "name" and "identifier" fields only, the others should not be able to be changed.

Reply:
Yes, it is doing the same - updating the name and rest OpenCDXIdentifier fields only (id, orgid, workspaceId)